src/routes/docs/products/functions/develop-locally/+page.markdoc

```diff
@@ -0,0 +1,95 @@
+
+We use Docker to replicate the production environment for the local deployment of functions. These can be executed locally with the `appwrite run functions` CLI command, which requires initializing a project with an `appwrite.json` file. The CLI also supports various other [CLI commands](/docs/tooling/command-line/commands).
+
+{% arrow_link href="/docs/tooling/command-line/functions" %}
+Learn more about deploying functions using CLI
+{% /arrow_link %}
+
+# Develop {% #develop %}
+
+Use the `appwrite run functions` command to develop your function locally.
+
+```sh
+appwrite run functions
+? Which function would you like to develop locally? My Awesome Function ("<FUNCTION_ID>")
+
+runtime   │ entrypoint  │ path                           │ commands     
+───────────┼──────────────┼───────────────────────────────────┼─────────────
+node-16.0 │ src/main.js │ functions/<FUNCTION_ID>        │ npm install
+
+ℹ Info: If you wish to change your local settings, update the appwrite.json file and rerun the 'appwrite run' command.
+♥ Hint: Permissions, events, CRON and timeouts dont apply when running locally.
+ℹ Info: Pulling Docker image ...
+♥ Hint: This may take a few minutes, but we only need to do this once.
+ℹ Info: Building function using Docker ...
+Preparing for build ...
+
+Building ...
+
+
+added 4 packages, and audited 5 packages in 2s
+
+
+1 package is looking for funding
+  run `npm fund` for details
+
+
+found 0 vulnerabilities
+
+Packing build ...
+
+Build finished.
+
+ℹ Info: Starting function using Docker ...
+♥ Hint: Function automatically restarts when you edit your code.
+✓ Success: Visit http://localhost:3000/ to execute your function.
+```
+
+This command helps you efficiently develop your Appwrite functions on your local machine. You can also hot reload your function code as you develop locally, so you will see your code changes instantly without waiting for it to build. When developing your Appwrite function locally, it will receive headers like when you develop a function in the Console.
+
+{% arrow_link href="/docs/products/functions/develop" %}
+Learn more about developing a function
+{% /arrow_link %}
+
+# Dynamic API keys {% #dynamic-api-keys %}
+
+You can use headers like dynamic API keys in your function, which give you access to your project services and allow you to operate without sessions. To configure your dynamic API key scopes, navigate to **Functions** > **Settings** > **Scopes**.
+
+{% arrow_link href="/docs/products/functions/develop" %}
+Learn more about dynamic API keys
+{% /arrow_link %}
+
+You can also impersonate a user when you develop a function locally.
+
+# Impersonate user {% #impersonate-user %}
+
+Impersonate a user using the `--userId <USER_ID>` option to select a user you want to use for testing. You can test if the user can perform a requested action, like creating a document.
+
+```sh
+appwrite run functions --userId "<USER_ID>"
```

Comment:
Isn't it `--user-id` rather than `--userId`?

Reply:
When I run `appwrite run functions -h`, it lists the option as `--userId`, so I followed the syntax listed here.

![Screenshot 2024-08-06 110239](https://github.com/user-attachments/assets/5236a611-2cae-414b-adbb-1c40467b3aa9)

Reply:
Looks like the commands and options were all updated to be kebab case. Please update them accordingly.